### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by vic778**
in this branch i have :

    Install CanCanCan in your project.
    Add a role column to the users table. Remember to use a migration for this.
    A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
        For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
    A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
        For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,24 +1,28 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @post = @user.posts.order(created_at: :desc).includes(:comments, :likes)
+    @user_posts = @user.posts.includes(:comments, :likes)
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb:6]: - [ ] according to the requirements you should Use CanCanCan for Post authorization in Post Controller, which might look something like below:
```
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Victoir, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] according to the below requirements you should implement destroy actions in the PostsController and in the CommentsController and there should be a delete button implemented that is only visible for authorized users.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

  - For that, you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

  - For that, you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @vic778  & @afizsavage , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mmsesay** [comment]
- [ ] As suggested by the previous reviewer, please implement a delete functionality for the PostsController and CommentsController and there should be a delete button implemented that is only visible for authorized users.

You can start off with declaring the delete method in this example:
```ruby
def destroy
   # the logic goes in here...
end
```

Please note:

 - A user can delete a post or comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

- Keep in mind that, you need to add the "Delete" button to the view and make sure that only authorized users can see it.

**Review by Emmyn5600 — APPROVED** [review]
Hi @vic778 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [ ] N/A

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**